src/config-file-watcher.ts

```diff
@@ -4,8 +4,8 @@ import * as vscodelc from 'vscode-languageclient/node';
 import {ClangdContext} from './clangd-context';
 import * as config from './config';
 
-export function activate(context: ClangdContext) {
-  if (config.get<string>('onConfigChanged') !== 'ignore') {
+export async function activate(context: ClangdContext) {
```

Comment:
The [call site](https://github.com/clangd/vscode-clangd/blob/2b21206d38c5dab72f7f9fd911b72e4827027e05/src/clangd-context.ts#L215) of this needs to be modified to use `await`.

Reply:
Fixed now. I thought it would be harmless to call registerFeature later, and it likely would be in this case, does not seem to be the expectation.

On the similar note, we probably want to move `switchSourceHeader.activate(this);` which also calls into registerFeature at the call site before `this.client.start();`